src/Utilities/SingleStackUtils/SingleStackUtils.jl

```diff
@@ -453,4 +454,258 @@ function dict_of_nodal_states(
     return OrderedDict(all_state_vars...)
 end
 
+# A container for holding various
+# global or point-wise states:
+struct States{P, A, GF}
+    prog::P
+    aux::A
+    diffusive::GF
+    hyperdiffusive::GF
+end
+
+"""
+    GridStack(
+            bl::BalanceLaw,
+            grid::DiscontinuousSpectralElementGrid,
+            prognostic,
+            auxiliary,
+            diffusive,
+            hyperdiffusive;
+            i = 1,
+            j = 1,
+            interp = true,
+        )
+
+A struct whose `iterate(::GridStack)` traverses
+the nodal stack and returns a NamedTuple of
+point-wise fields (`Vars`).
+
+# Example
+```julia
+for state_local in GridStack(
+        bl,
+        grid,
+        prognostic, # global field along nodal stack
+        auxiliary,
+        diffusive,
+        hyperdiffusive
+    )
+prog = state_local.prog # point-wise field along nodal stack
+end
+```
+
+## TODO: Make `prognostic`, `auxiliary`, `diffusive`, `hyperdiffusive` optional
+
+# Arguments
+ - `bl` the balance law
+ - `grid` the discontinuous spectral element grid
+ - `prognostic` the global prognostic state
+ - `auxiliary` the global auxiliary state
+ - `diffusive` the global diffusive state (gradient-flux)
+ - `hyperdiffusive` the global hyperdiffusive state
+ - `i,j` the `i,j`'th nodal stack (in the horizontal directions)
+ - `interp` a bool indicating whether to
+    interpolate the duplicate Gauss-Lebotto
+    points at the element faces.
+
+!!! warn
+    Before iterating, the data is transferred from the
+    device (GPU) to the host (CPU), as this is intended
+    for debugging / diagnostics usage.
+"""
+struct GridStack{N, BL, G, S, VR, TI, TJ, CI, IN}
+    bl::BL
+    grid::G
+    states::S
+    vrange::VR
+    i::TI
+    j::TJ
+    cart_ind::CI
+    interp::IN
+    function GridStack(
+        bl::BalanceLaw,
+        grid::DiscontinuousSpectralElementGrid,
+        prognostic,
+        auxiliary,
+        diffusive,
+        hyperdiffusive;
+        i = 1,
+        j = 1,
+        interp = true,
+    )
+        states = States(prognostic, auxiliary, diffusive, hyperdiffusive)
+        vrange = 1:size(prognostic, 3)
+        N = polynomialorders(grid)
+        dim = dimensionality(grid)
+        Nq = N .+ 1
+        Nqk = dim == 2 ? 1 : Nq[dim]
+        Nk = dim == 2 ? 1 : N[dim]
```

Comment:
Use [`basic_grid_info`](https://github.com/CliMA/ClimateMachine.jl/blob/master/src/Numerics/DGMethods/DGModel.jl#L130) (and [`basic_topology_info`](https://github.com/CliMA/ClimateMachine.jl/blob/master/src/Numerics/Mesh/Topologies.jl#L1333) if needed) instead.

Reply:
Will do, thanks!